tests/testthat/test-get_source_expressions.R

```diff
@@ -292,8 +290,7 @@ skip_if_not_installed("patrick")
 #   fail on files where the XML content is xml_missing;
 #   the main linter test files provide more thorough
 #   evidence that things are working as intended.
-bad_source <- withr::local_tempfile()
-writeLines("a <- 1L\nb <- 2L", bad_source)
+bad_source <- withr::local_tempfile(lines = "a <- 1L\nb <- 2L")
```

Comment:
This will add a terminal newline, IINM.

Reply:
Are you sure about that? Am I missing something here?

``` r
bad_source1 <- withr::local_tempfile()
writeLines("a <- 1L\nb <- 2L", bad_source1)

bad_source2 <- withr::local_tempfile(lines = "a <- 1L\nb <- 2L")

waldo::compare(readLines(bad_source1), readLines(bad_source2))
#> v No differences
```

<sup>Created on 2022-06-19 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

Reply:
Ah, sorry, `writeLines()` adds the final newline implicitly.
`cat()` wouldn't have.
But then we could use `lines = c("a <- 1L", "b <- 2L")` for a little improved readability.

Reply:
Okay, done.